tools/setup/emoji/emoji_setup_utils.py

```diff
@@ -0,0 +1,179 @@
+    'flipper', # dolphin
+]
+
+## functions that take in a list of names at a codepoint and return a subset to exclude
+
+def blacklisted(names):
+    # type: (List[str]) -> List[str]
+    return [name for name in names if name in blacklisted_names]
+
+# 1 letter names don't currently show up in our autocomplete. Maybe should
+# change our autocomplete so that a whitelist of letters do, like j (for joy), x, etc
+# github uses a, ab, etc. instead of a_button, slack doesn't have any of the [letter]_buttons
+def one_lettered(names):
+    # type: (List[str]) -> List[str]
+    if len(names) == 1:
+        return []
+    return [name for name in names if len(name) == 1]
+
+# If it is an ideograph (or katakana, but we'll probably deal with that
+# differently after 1.5), remove any names that don't have
+# ideograph/katakana in them
+def ideographless(names):
+    # type: (List[str]) -> List[str]
+    has_ideographs = ['ideograph' in name.split('_') or
+                      'katakana' in name.split('_') for name in names]
+    if True not in has_ideographs:
+        return []
+    return [name for name, has_ideograph in zip(names, has_ideographs) if not has_ideograph]
+
+# subsumed by longer, but still useful for breaking up a hand review of the
+# blacklist decisions
+def word_superset(names):
+    # type: (List[str]) -> List[str]
+    bags_of_words = [frozenset(name.split('_')) for name in names]
+    bad_names = set()
+    for i, j in permutations(range(len(names)), 2):
+        if bags_of_words[i] < bags_of_words[j]:
+            bad_names.add(names[j])
+    return list(bad_names)
+
+# subsumed by longer, but still useful for breaking up a hand review of the
+# blacklist decisions
+def superstring(names):
+    # type: (List[str]) -> List[str]
+    bad_names = set()
+    for name1, name2 in permutations(names, 2):
+        if name2[:len(name1)] == name1:
+            bad_names.add(name2)
+    return list(bad_names)
+
+def longer(names):
+    # type: (List[str]) -> List[str]
+    lengths = [len(name) for name in names]
+    min_length = min(lengths)
+    return [name for name, length in zip(names, lengths) if length > min_length]
+
+def update_names_at_codepoint(func, codepoint, names):
+    # type: (Callable[[List[str]], List[str]], str, List[str], bool) -> None
+    bad_names = func(names)
+    good_names = [name for name in names if name not in bad_names]
+
+def emoji_names_for_picker(emoji_map):
+    # type: (Dict[Text, Text]) -> List[str]
+    codepoint_to_names = defaultdict(list) # type: Dict[str, List[str]]
+    for name, codepoint in emoji_map.items():
+        codepoint_to_names[codepoint].append(str(name))
+
+    # blacklisted must come first, followed by {one_lettered, ideographless}
+    for func in [blacklisted, one_lettered, ideographless, word_superset, superstring, longer]:
+        for codepoint, names in codepoint_to_names.items():
+            codepoint_to_names[codepoint] = [name for name in names if name not in func(names)]
```

Comment:
Add a comment explaining that each function returns which ones are to be `REMOVED` (not kept) from the set.